cli/compose/convert/service.go

```diff
@@ -214,18 +214,21 @@ func convertServiceNetworks(
 		if !ok && networkName != defaultNetwork {
 			return nil, errors.Errorf("undefined network %q", networkName)
 		}
-		var aliases []string
-		if network != nil {
-			aliases = network.Aliases
-		}
 		target := namespace.Scope(networkName)
 		if networkConfig.External.External {
 			target = networkConfig.External.Name
 		}
-		nets = append(nets, swarm.NetworkAttachmentConfig{
-			Target:  target,
-			Aliases: append(aliases, name),
-		})
+		netAttachConfig := swarm.NetworkAttachmentConfig{
+			Target: target,
+		}
+		if container.NetworkMode(target).IsUserDefined() {
```

Comment:
Yes, you would expect this behaviour because you're familiar with it :)

A user has no idea of the limitations of `host`, and silently ignoring the aliases will lead to lots of confusion. Here's what I get with the latest docker-compose release:

```
ERROR: for examples_web_1  network-scoped alias is supported only for containers in user defined networks
```

We should keep the same behaviour and let the server error out for us. So I think we should revert this part of the diff.  

If you agree, I can take care of it.